### ISSUE TITLE
Ingress Address is not  getting populated.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#12169]
Rancher server version - 2.0 build from master 

Setup - Cluster with 3 DO nodes
Steps to reproduce the problem:
Create a workload of type daemonset.
Create an Ingress with rule pointing to this workload.
Ingress is functional from all the 3 node ip address and is able to reach all 3 targets.

When viewing this ingress from kubectl , address is not being populated in the ingress object created.
```
$ ./kubectl get ingress
NAME      HOSTS     ADDRESS   PORTS     AGE
mytest1   my.com              80        15m
```

**Comment by orangedeng** [comment]
Because we set a specific labels for each pod in  `https://github.com/rancher/rancher/blob/master/pkg/controllers/user/networkpolicy/podhandler.go#L51`, nginx controller pod can not get its peers. And then it can only get its node address instead of all node's. After discuss with @alena1108 , we decide to fork `https://github.com/kubernetes/ingress-nginx` and fix it in our ingress controller.

**Comment by galal-hussein** [comment]
@orangedeng rke side is for this issue is merged, i added address annotations to each node

**Comment by sangeethah** [comment]
Tested with - v2.0.0-beta3-rc1 build.

Setup - Cluster with  DO node.

Steps to reproduce the problem:
Create a workload of type daemonset.
Create an Ingress with rule pointing to this workload.
Ingress is functional from host ip address.
Still dont see the address injected in the ingress

```
> kubectl get ingress
NAME      HOSTS       ADDRESS   PORTS     AGE
hello     test1.com             80        34m
>
```

**Comment by sangeethah** [comment]
For the first ingress created in default namespace , i see the address being injected eventually , though I had not seen for atleast 15 minutes from from time of ingress creation.
For the next next few ingresses I created in the same Default namespace , I was able to see the address show up in ingress but after a delay of few seconds.

When i create a  ingress in a newspace , again I saw a delay of about 1 minute before the ip gets injected in the ingress.
